cirq/aqt/aqt_sampler.py

```diff
@@ -0,0 +1,226 @@
+import json
+import time
+import uuid
+from typing import Iterable, List, Union
+
+import numpy as np
+from requests import put  # TODO: Specify as a dependency
+from cirq import XPowGate, YPowGate, XXPowGate
+from cirq import circuits, schedules, study, Sampler, resolve_parameters
+from cirq.ops.eigen_gate import EigenGate
+from cirq.study.resolver import ParamResolver
+from cirq.study.sweeps import Sweep
+from cirq.aqt.aqt_simulator import AQTSimulator
+
+Sweepable = Union[ParamResolver, Iterable[ParamResolver], Sweep,
+                  Iterable[Sweep]]
+"""Samplers for the AQT ion trap device"""
+
+
+def get_op_string(op_obj: EigenGate):
+    """Find the string representation for a given gate
+    Params:
+        op_obj: Gate object, out of: XXPowGate, XPowGate, YPowGate"""
+    if isinstance(op_obj, XXPowGate):
+        op_str = 'MS'
+    elif isinstance(op_obj, XPowGate):
+        op_str = 'X'
+    elif isinstance(op_obj, YPowGate):
+        op_str = 'Y'
+    else:
+        raise RuntimeError('Got unknown gate:', op_obj)
+    return op_str
+
+
+class AQTSampler(Sampler):
+    """Sampler for the AQT ion trap device
+    This sampler connects to the AQT machine and
+    runs a single circuit or an entire sweep remotely
+    """
+
+    def _run_api(
+            self,
+            circuit: circuits.Circuit,
+            param_resolver: Sweepable,
+    ) -> str:
+        """
+        Args:
+            circuit: Circuit to be run
+            param_resolver: Param resolver for the
+
+        Returns:
+            json formatted string of the sequence
+        """
+
+        seq_list = []
+        circuit = resolve_parameters(circuit, param_resolver)  # type: ignore
+        # TODO: Check if circuit is empty
+        for op in circuit.all_operations():
+            qubits = [obj.x for obj in op.qubits]  # type: ignore
+            op_str = get_op_string(op.gate)  # type: ignore
+            seq_list.append([op_str, op.gate.exponent, qubits])  # type: ignore
+        json_list = json.dumps(seq_list)
+        return json_list
+
+    def _send_json(
+            self,
+            json_str: str,
+            id_str: Union[str, uuid.UUID],
+            remote_host: str = 'http://localhost:5000',
+            access_token: str = '',
+            repetitions: int = 1,
+            no_qubit: int = 1,
+    ):
+        """Sends the json string to the remote AQT device
+        Args:
+            json_str: json representation of the circuit
+            id_str: Unique id of the datapoint
+            remote_host: address of the remote device
+            repetitions: Number of repetitions
+            no_qubit: Number of qubits present in the device
+
+        Returns:
+            measurement results as an array of boolean
+        """
+        data = put(remote_host,
+                   data={
+                       'data': json_str,
+                       'id': id_str,
```

Comment:
Why is the access token not included in this first call?

Merge this call into the call within the loop by using an `if finished: break` at the end.